[USDU-260] Fix 'ExportSelectedWithChildren' on Mac

vickycl opened the pull request:
## Purpose of this PR

**Ticket/Jira #:** [USDU-260](https://jira.unity3d.com/browse/USDU-260)

MacOS doesn't support passing multiple file extensions to EditorUtility.SaveFilePanel and so was appending all of them. This changes the 'Export Selected With Children' button on non-Windows platform to default to .usd, and adds a 'Export Selected With Children as USDA' option for the alternative.

![Screenshot 2023-01-27 at 15 19 09](https://user-images.githubusercontent.com/45685026/215122143-6a56e3fe-eb88-4f2e-8a0b-95600fbc1f2c.png)



## Testing

**Functional Testing status:**

Manually tested that the bug no longer reproduces, and looked at the resulting .usd in USDView to confirm the output was as expected.

**Performance Testing status:**

No anticipated performance impact.

## Overall Product Risks
<!-- See Testing Status, Complexity and Halo Effect for your PR](https://confluence.unity3d.com/display/DEV/Testing+Status%2C+Complexity+and+Halo+Effect+for+your+PR). -->

**Complexity:** low
<!-- (Minimal / Low / Medium / High) -->

**Halo Effect:** lox
<!-- (Minimal / Low / Medium / High) -->

## Additional information

**Note to reviewers:**

<!-- Info per person for what to focus on, or historical info to understand who have previously reviewed and coverage. Help them get context. -->

**Reminder:**
<!-- Things to remember to do before finalizing this PR. -->
- [ ] Add entry in CHANGELOG.md _(if applicable)_

vickycl committed:
Fix 'ExportSelectedWithChildren' on Mac

MacOS doesn't support passing multiple file extensions to EditorUtility.SaveFilePanel and so was appending all of them.

vickycl committed:
Better fix that leaves Windows behaviour the same.

SimonBoorer commented:
You could also add two options on non-Windows platforms (e.g. Export USD ASCII File, Export USD Crate File)

vickycl committed:
Add additional save menu options for non-windows editor platforms.

vickycl committed:
Removed the standard .usd option

Giving a USDC extension on Mac becomes .usd anyway, so removed the other option to avoid confusion

vickycl committed:
Small UI change

vickycl committed:
Remove USDC option, as the default export of .usd is binary crate anyway

vickycl committed:
Merge branch 'dev' into usdu-260-fix-usd-mac-export

SimonBoorer commented on package/com.unity.formats.usd/Editor/Scripts/Behaviors/UsdMenu.cs:
```diff
@@ -23,6 +23,7 @@ namespace Unity.Formats.USD
     public static class UsdMenu
     {
         [MenuItem("USD/Export Selected with Children", true)]
+        [MenuItem("USD/Export Selected with Children as USDA", true)]
```
Is it OK that this is defined on Windows but the actual `MenuExportSelectedWithChildrenToUSDA` is not?

vickycl replied:
From first read of the docs it sounded like it was fine but looking at the code it seems like it could be a weird edge case. Although it works as expected on Windows currently I'm not confident it won't break in future. I will change it :)

vickycl committed:
Remove the MenuItem validation on Windows where it's not used

Although it works as expected on Windows currently I'm not confident it won't break in future as it seems like it could be a weird edge case.

vickycl committed:
Merge branch 'usdu-260/save-menu-options' into usdu-260-fix-usd-mac-export

vickycl committed:
Merge branch 'dev' into usdu-260-fix-usd-mac-export
